packages/functional-tests/pages/products/index.ts

```diff
@@ -44,10 +47,30 @@ export class SubscribePage extends BaseLayout {
     await paypalWindow.click('button[id=consentButton]');
   }
 
+  async addExpiredCoupon() {
+    await this.page.click('[data-testid="coupon-input"]');
+    await this.page.fill('[data-testid="coupon-input"]', 'autoexpired');
+    await this.page.click('[data-testid="coupon-button"]');
+  }
+
+  async addInvalidCoupon() {
+    await this.page.click('[data-testid="coupon-input"]');
+    await this.page.fill('[data-testid="coupon-input"]', 'autoinvalid');
+    await this.page.click('[data-testid="coupon-button"]');
+  }
```

Comment:
nit: Not sure if we want to DRY this out a bit and add locators for `[data-testid="coupon-input"]`, etc since they seem to be duplicated a bit now.